### DISCUSSION TITLE
prisma with supabase and connection pooling

### DISCUSSION THREAD

**Discussion by empato-limited** [prisma/prisma#17109] [Q&A] [answered]
I am trying to connect supabase to a remix app ... will deploy to vercel so i need to use connection pooling
schema.prisma looks like this:
```
datasource db {
  provider = "postgresql"
  url      = env("DATABASE_URL")
  shadowDatabaseUrl = env("SHADOW_DATABASE_URL")
}

generator client {
  provider = "prisma-client-js"
}

model Extension {
  id    String @id @default(cuid())
}
```

.env:
```
DATABASE_URL=postgres://aaa:aaa@db.xxx.supabase.co:6543/postgres?pgbouncer=true&connection_limit=1
SHADOW_DATABASE_URL=postgres://aaa:aaa@db.xxx.supabase.co:6543/postgres_shadow
```

when i run ```js
prisma migrate dev --name init 
```
i get 
```
Error: db error: FATAL: bouncer config error
   0: migration_core::state::DevDiagnostic
             at migration-engine/core/src/state.rs:264
```

**Comment by nurul3101** [comment, marked as answer]
Hello @empato-limited 👋 

Can you try running the migrate command on connecting string ending with port 5432?
Based on this [comment](https://github.com/prisma/prisma/issues/9576#issuecomment-946578235), migrate command should work as expected on non pooled connection string.

**Comment by empato-limited** [comment]
that was exactly it ... thank you .... for future reference ... for those using supabase i will below the steps i took to fix this because I do not think that was my only mistake...

1) initially i created postgres_shadow from the supabase web query tool ... i do not know think that tool runs as the postgres because afterwards i connected manually with a desktop app and i (the postgres user was not the owner of that db)
2) decided to start from scratch ... but there was another thing wrong
3) the db endpoints are not correct ... ended up using:
```
DATABASE_URL=postgres://aaa:aaa@db.xxx.supabase.co:6543/postgres?pgbouncer=true&connection_limit=1
SHADOW_DATABASE_URL=postgres://aaa:aaa@db.xxx.supabase.co:5432/postgres_shadow
```
so the difference is in the ports ... now to be fair i did not deploy it yet to vercel ... but for now at least i could run every prisma command

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
